tests/back/dataloaders/test_dashai_dataset.py

```diff
@@ -239,3 +244,32 @@ def test_save_to_disk_and_load():
     assert dashai_datasetdict["train"].outputs_columns == outputs_columns
     assert dashai_datasetdict["validation"].outputs_columns == outputs_columns
     assert dashai_datasetdict["test"].outputs_columns == outputs_columns
+
+
+def test_update_column_types():
+    dataset = split_dataset()
+    modify_data = {
+        "SepalLengthCm": ColumnUpdateItemParams(type="Value", dtype="string"),
+        "SepalWidthCm": ColumnUpdateItemParams(type="Value", dtype="float64"),
+        "PetalLengthCm": ColumnUpdateItemParams(type="Value", dtype="string"),
+        "PetalWidthCm": ColumnUpdateItemParams(type="Value", dtype="float64"),
+        "Species": ColumnUpdateItemParams(type="Value", dtype="string"),
+    }
+
+    save_dataset(dataset, "tests/back/dataloaders/dashaidataset")
+    dataset_update = update_column_types(
+        "tests/back/dataloaders/dashaidataset", modify_data
+    )
+    shutil.rmtree("tests/back/dataloaders/dashaidataset", ignore_errors=True)
+
+    new_features = dataset_update["train"].features
```

Comment:
Also, check if the dict does not have any more (or less) columns than the expected:

```suggestion
    assert list(new_features.keys()) == ["SepalLengthCm", "SepalWidthCm", ["PetalLengthCm",  "PetalWidthCm", "Species"]
    new_features = dataset_update["train"].features
```